service/utils/tds.py

```diff
@@ -61,55 +102,72 @@ def cleanup_job_dir(job_id):
     shutil.rmtree(path)
 
 
-def fetch_model(model_config_id, tds_api, config_endpoint, job_id):
+def fetch_model(model_config_id, job_id):
     job_dir = get_job_dir(job_id)
     logging.debug(f"Fetching model {model_config_id}")
-    url_components = [tds_api, config_endpoint, model_config_id]
-    model_url = ""
-    for component in url_components:
-        model_url = urllib.parse.urljoin(model_url, component)
-    model_response = requests.get(model_url)
+
+    model_url = TDS_URL + TDS_CONFIGURATIONS + "/" + model_config_id
+
+    model_response = tds_session().get(model_url)
     if model_response.status_code == 404:
         raise HTTPException(status_code=404, detail="Model not found")
+
     amr_path = os.path.join(job_dir, f"./{model_config_id}.json")
     with open(amr_path, "w") as file:
         json.dump(model_response.json()["configuration"], file)
     return amr_path
 
 
-def fetch_dataset(dataset: dict, tds_api, job_id):
+def fetch_dataset(dataset: dict, job_id):
     import pandas as pd
 
     job_dir = get_job_dir(job_id)
     logging.debug(f"Fetching dataset {dataset['id']}")
     dataset_url = (
-        f"{tds_api}/datasets/{dataset['id']}/"
+        f"{TDS_URL}{TDS_DATASETS}/{dataset['id']}/"
         f"download-url?filename={dataset['filename']}"
     )
-    response = requests.get(dataset_url)
+    response = tds_session().get(dataset_url)
+
+    # Small hack to rename mapping, namely timestamp => Timestamp if timestamp exist as a value
```

Comment:
What is this hack for?

Reply:
It seems like ciemss needs a specific format regardless of the mappings provided.